distribution/tools/keystore-cli/src/main/java/org/elasticsearch/common/settings/AddFileKeyStoreCommand.java

```diff
@@ -49,43 +49,45 @@
         // jopt simple has issue with multiple non options, so we just get one set of them here
         // and convert to File when necessary
         // see https://github.com/jopt-simple/jopt-simple/issues/103
-        this.arguments = parser.nonOptions("setting [filepath]");
+        this.arguments = parser.nonOptions("(setting path)+");
     }
 
     @Override
     protected void executeCommand(Terminal terminal, OptionSet options, Environment env) throws Exception {
-        List<String> argumentValues = arguments.values(options);
+        final List<String> argumentValues = arguments.values(options);
         if (argumentValues.size() == 0) {
             throw new UserException(ExitCodes.USAGE, "Missing setting name");
         }
-        String setting = argumentValues.get(0);
+        if (argumentValues.size() % 2 != 0) {
```

Comment:
Don't we need to also ensure size > 1? Otherwise I think we silently ignore giving the add-file command with no arguments?

Reply:
That's handled on the previous `if`? If `size <= 0` then it must be `== 0` and we bail?

Reply:
The `if` currently on line 58 of this diff.

Reply:
I can't believe I missed that. :shame: